crates/core/asset/src/balance.rs

```diff
@@ -29,17 +28,50 @@ use decaf377::{r1cs::ElementVar, Fq, Fr};
 use imbalance::Imbalance;
 
 use self::commitment::BalanceCommitmentVar;
+use penumbra_proto::{penumbra::core::asset::v1 as pb, DomainType};
 
 /// A `Balance` is a "vector of [`Value`]s", where some values may be required, while others may be
 /// provided. For a transaction to be valid, its balance must be zero.
-#[serde_as]
 #[derive(Clone, Eq, Default, Serialize, Deserialize)]
+#[serde(try_from = "pb::Balance", into = "pb::Balance")]
 pub struct Balance {
     negated: bool,
-    #[serde_as(as = "Vec<(_, _)>")]
     balance: BTreeMap<Id, Imbalance<NonZeroU128>>,
 }
 
+/* Protobuf impls */
+impl DomainType for Balance {
+    type Proto = pb::Balance;
+}
+
+impl TryFrom<pb::Balance> for Balance {
+    type Error = anyhow::Error;
+
+    fn try_from(v: pb::Balance) -> Result<Self, Self::Error> {
+        let mut balance_map = BTreeMap::new();
+
+        for imbalance_value in v.balance.into_iter().map(TryInto::try_into) {
+            let value: Value = imbalance_value?;
+            let amount = NonZeroU128::new(value.amount.into())
+                .ok_or_else(|| anyhow::anyhow!("amount must be non-zero"))?;
+
+            let imbalance = Imbalance::Provided(amount); // todo: fix this placeholder
```

Comment:
comment: `Values` cannot be negative, a negative `Balance` is formed by negating a balance derived from a value. How do we calculate the Imbalance and differentiate between provided and required balances?

Reply:
I think the modeling suggested by henry here: https://github.com/penumbra-zone/penumbra/pull/4943/files#r1855950445 takes care of that

Reply:
imbalance is now determined based on the `negated` flag for the individual `SignedValue` pairs.